rust/connlib/tunnel/src/gateway.rs

```diff
@@ -305,6 +351,22 @@ impl GatewayState {
         Ok(())
     }
 
+    pub(crate) fn setup_dns_resource_nat(
+        &mut self,
+        client: ClientId,
+        domain: DomainName,
+        resolved_ips: Vec<IpAddr>,
+        proxy_ips: Vec<IpAddr>,
+        now: Instant,
+    ) -> anyhow::Result<()> {
+        let peer = self.peers.get_mut(&client).context("Unknown client")?;
+        let resource_id = todo!("Find resource ID by domain name and verfiy it is a sub-domain");
```

Comment:
The resource ID can come from the client. We can then just look up and check the domain.